Program.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using Microsoft.VisualBasic.FileIO;
+using System.Collections.Generic;
+
+namespace SupportBank
+{
+    class Program
+    {
+        static Dictionary<string, Account> ReadFile() {
+
+            var accounts = new Dictionary<string, Account>();
+
+            var parser = new TextFieldParser(@"Transactions2014.csv");
+            parser.TextFieldType = FieldType.Delimited;
+            parser.SetDelimiters(",");
+
+            if (!parser.EndOfData)
+                parser.ReadLine();
+
+            while (!parser.EndOfData)
+            {
+                var fields = parser.ReadFields();
+                var newTrans = new Transaction(fields);
+
+                foreach (string name in fields[1..3])
+                {
+                    if (!accounts.ContainsKey(name))
+                    {
+                        accounts.Add(name, new Account(name));
+                    }
+                    accounts[name].AddTransaction(newTrans);
+                }
+            }
+            return accounts;
+        }
+
+        static string ReadConsole(string prompt)
+        {
+            string answer = null;
+            do
+            {
+                Console.Write(prompt);
+                try
+                {
+                    answer = Console.ReadLine();
+                }
+                catch
+                {
+                }
+            } while (string.IsNullOrEmpty(answer));
+
+            return answer;
+        }
+
+        static void ListAll(Dictionary<string, Account> accounts)
+        {
+            foreach (string name in accounts.Keys)
+            {
+                Console.WriteLine(accounts[name].GetSummary());
+            }
+        }
+
+        static void ListAccount(Dictionary<string, Account> accounts, string name)
+        {
+            foreach (Transaction trans in accounts[name].transactions)
+            {
+                Console.WriteLine(trans.GetSummary());
+            }
+        }
+
+        static bool ExecuteCommand(Dictionary<string, Account> accounts)
+        {
+            string input;
+
+            do
+            {
+                input = ReadConsole("Input a command: ");
+            } while (!input.StartsWith("List "));
+
+            string param = input.Substring(5);
+
+            switch (param)
+            {
+                case "All":
+                    ListAll(accounts);
+                    break;
+                default:
+                    if (!accounts.ContainsKey(param))
+                        Console.WriteLine("Account not found");
+                    ListAccount(accounts, param);
+                    break;
+            }
+
+            return true;
```

Comment:
Does this ever not return `true`?

Reply:
Writing the loop this way was to allow future commands to exit the program by returning false. I should have included a return true after executing the existing commands to make this more obvious.

Reply:
So this is the kind of future proofing I would avoid 😛  As it makes the code as it is more confusing. Also I wouldn't expect a function called `ExecuteCommand` to return a boolean, it's not clear what that boolean represents. So if you did want a future command to exit the program then I'd suggest slightly refactoring this anyway 🙂  Does that make sense?

Reply:
Yes, thanks for the explanation!